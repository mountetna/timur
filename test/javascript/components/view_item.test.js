import React from 'react';
import {render, screen, waitFor, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ViewItem from '../../../lib/client/jsx/components/browser/view_item';

describe('ViewItem', () => {
  it('renders without a regex', async () => {
    const {asFragment} = render(
      <ViewItem
        item={{
          name: 'attribute1',
          type: 'markdown',
          text: 'This is a test'
        }}
        record_name='PROJ1-HS1'
      />
    );

    await waitFor(() => screen.getByText('This is a test'));

    expect(screen.queryByText(/This is a test/)).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with a regex that matches the record_name', async () => {
    const {asFragment} = render(
      <ViewItem
        item={{
          name: 'attribute1',
          type: 'markdown',
          text: 'This is a test',
          regex: 'PROJ1'
        }}
        record_name='PROJ1-HS1'
      />
    );

    await waitFor(() => screen.getByText('This is a test'));

    expect(screen.queryByText(/This is a test/)).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  it('does not render if regex false', () => {
    const {container} = render(
      <ViewItem
        item={{
          name: 'attribute1',
          type: 'markdown',
          text: 'This is a test',
          regex: '.*SCG.*'
        }}
        record_name='PROJ1-HS1'
      />
    );
    expect(container.innerHTML).toHaveLength(0);
  });
});
