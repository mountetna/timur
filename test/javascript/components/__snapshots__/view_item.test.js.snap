// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewItem renders with a regex that matches the record_name 1`] = `
<DocumentFragment>
  <div
    class="view_item"
  >
    <div
      class="item_name"
    >
      attribute1
    </div>
    <div
      class="item_view"
    >
      <div
        class="item"
      >
        <div
          class="markdown"
        >
          <p>
            This is a test
          </p>
          

        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ViewItem renders without a regex 1`] = `
<DocumentFragment>
  <div
    class="view_item"
  >
    <div
      class="item_name"
    >
      attribute1
    </div>
    <div
      class="item_view"
    >
      <div
        class="item"
      >
        <div
          class="markdown"
        >
          <p>
            This is a test
          </p>
          

        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
